tests/unittest/_torch/auto_deploy/unit/singlegpu/custom_ops/torch_attention_reference.py

```diff
@@ -0,0 +1,203 @@
+"""Torch attention reference implementations for testing.
+
+This module provides clean reference implementations using the torch backend
+that can be used across all attention operation test files to eliminate
+code duplication and ensure consistency.
+"""
+
+import torch
+
+import tensorrt_llm._torch.auto_deploy  # noqa: F401
+from tensorrt_llm._torch.auto_deploy.custom_ops.torch_backend_attention import (
+    torch_backend_mha_with_cache,
```

Comment:
I am surprised the torch custom ops decorator didn't complain the usage like this. I believe the convention is to call the function via the custom op rather than call the function definition directly